include/component.hxx

```diff
@@ -200,6 +200,25 @@ T getNoBoundary(const Options& option, const std::string& location = "") {
   getNoBoundary<Type>(option)
 #endif
 
+/// Check whether value is valid, returning true
+/// if invalid i.e contains non-finite values
+template<typename T>
+bool hermesDataInvalid(const T& value) {
+  return false; // Default
+}
+
+/// Check Field3D values.
+/// Doesn't check boundary cells
+template<>
```

Comment:
Quick question: Why do you use a template here rather than a inline function?

Reply:
I think because it's called from the `set` function, which itself is a template. It only has an implementation for `Field3D` but may be instantiated for other types.